daemon/logger/loggerutils/rotatefilewriter.go

```diff
@@ -90,20 +96,125 @@ func rotate(name string, maxFiles int) error {
 	if maxFiles < 2 {
 		return nil
 	}
-	for i := maxFiles - 1; i > 1; i-- {
-		toPath := name + "." + strconv.Itoa(i)
-		fromPath := name + "." + strconv.Itoa(i-1)
+
+	for i := maxFiles - 1; i > 2; i-- {
+		toPath := name + "." + strconv.Itoa(i) + ".gz"
+		fromPath := name + "." + strconv.Itoa(i-1) + ".gz"
 		if err := os.Rename(fromPath, toPath); err != nil && !os.IsNotExist(err) {
 			return err
 		}
 	}
 
+	if _, err := os.Stat(name + ".1"); err == nil && maxFiles > 2 {
+		if err := os.Rename(name+".1", name+".2"); err != nil {
+			return err
+		}
+
+		if err := compressJSONFile(name + ".2"); err != nil {
```

Comment:
It would be nice if we can handle compression asynchronously from log writes.
We'd probably have to have a write lock and a rotation lock, basically the next rotation would be blocked until compression is completed but writing can continue up until it needs to rotate again.